File: 1328-break-a-palindrome/1328-break-a-palindrome.cpp
class Solution {
public:
    string breakPalindrome(string palindrome) {
        int n = palindrome.length();
        if (n == 1) {
            string res;
            return res;
        }
        vector<char> p;
        for (int i=0; i<n; i++) {
            p.push_back(palindrome[i]);
        }
        for (int i=0; i<(n/2); i++) {
            if (palindrome[i] != 'a') {
                palindrome[i] = 'a';
                string res;
                for (int i=0; i<n; i++) {
                    res += palindrome[i];
                }
                return res;
            }
        }
        char last_char = (int)palindrome[n-1] + 1;
        palindrome[n-1] = last_char;
        string res;
        for (int i=0; i<n; i++) {
            res += palindrome[i];
        }
        return res;
    }
};

File: 1328-break-a-palindrome/1328-break-a-palindrome.py
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        n = len(palindrome)
        if n == 1:
            return ""
        palindrome = list(palindrome)
        for i in range(n // 2):
            if palindrome[i] != "a":
                palindrome[i] = "a"
                return "".join(palindrome)
        palindrome[-1] = chr(ord(palindrome[-1]) + 1)
        return "".join(palindrome)
